jOOQ-codegen/src/main/java/org/jooq/util/JavaGenerator.java

```diff
@@ -4439,43 +4443,45 @@ protected void printTableJPAAnnotation(JavaWriter out, TableDefinition table) {
                 out.print(scala ? ")" : "}");
             }
 
-            StringBuilder sb2 = new StringBuilder();
-            String glue2 = "\n";
+            if (generateJpaVersion() != null && generateJpaVersion().supportedIndex()) {
```

Comment:
If the configured version is null, then the expression should yield true. Thus:

```java
if (generateJpaVersion() == null || ...) {
```

I'll fix this as well...